keymanagementrequirements.md

```diff
@@ -0,0 +1,138 @@
+
+    The certificate of the signing key or the timestamp key (whichever is longer) expires
+    The signed content indicates a expiry time
+
+In this section, the #2 scenario is focused.
+Scenarios
+
+The main scenario for using a configurable signature expiry is for the publisher to indicate how long they plan to maintain the signature on the artifact for. For example if a publisher signs an artifact with an expiry for 6 months, they are indicating that if within 6 months they identfy a reason that deployers should no longer trust the artifact they will rescind the signature.
+
+An added benefit is to defend against freeze attack. Since the signature expires in a short time although the signing key can live longer, the verifiers have to obtain the latest signature to verify, which implies that the verifies have access to the latest content.
+
+    Freeze Attack Similar to a replay attack, a freeze attack works by providing metadata that is not current. However, in a freeze attack, the attacker freezes the information a client sees at the current point in time to prevent the client from seeing updates, rather than providing the client older versions than the client has already seen. As with replay attacks, the attacker’s goal is ultimately to compromise a client who has vulnerable versions of packages installed. A freeze attack may be used to prevent updates in addition to having an installed package be out of date.
+
+### Transparent Root key auto rotation
+Root keys form the basis of hierarchical trust systems, where intermediate and leaf keys chain back to a root key, and leaf keys are used to generate signatures. Consumers of signed artifacts associate a baseline level of trust with signatures that chain to roots they trust. In the scenario that a root key is no longer trusted, all signed artifacts chaining back to the root are no longer trusted. The conditions which render the root keys to be no longer trusted, are disclosed to the customer through some other channel (CVE, internal security audit, etc.). Safe rotation of keys require a overlap period where both new and old root keys are trusted. This is always not possible, such as when a root key is compromised.
+
+#### Scenarios
+1. A public root associated with a key a Publisher uses to sign artifacts is compromised. The access to root key is compromised, rather than the root key itself being stolen, allowing an attacker to create new intermediate and leaf trusted keys.
+2. An internal root used by an organization is due to expire. A new root is created and rotated before the existing root expires.
+3. An internal root used by an organization is not stored securely and the private key is lost. A new root is created to continue operations.
+4. A cryptographic algorithm is deprecated, or a compliance standard requires a customer to upgrade their key strength. The customer wants to safely rotate a root key without rendering existing signatures invalid and disrupting operations.
```

Comment:
~~customer~~ - user

Reply:
@SteveLasker could you clarify, is the suggestion to use *user* instead of *customer* in the entire doc? I plan to replace them with specific personas we defined in the next rev, *Publisher Admin* would be appropriate here.

Reply:
@gokarmm, yeah. In the OSS community, I'm just trying to avoid _customer_ references as not all users are _customers_, but all _customers_ are _users_.

Using a persona name could help, but I imagine there are some cases where multiple personas would apply. So, just suggesting _user_ would be a bit more generic where multiple personas may apply